Source/FieldSolver/FiniteDifferenceSolver/MacroscopicProperties/MacroscopicProperties.cpp

```diff
@@ -161,30 +119,108 @@ MacroscopicProperties::InitData ()
     amrex::Print() << "we are in init data of macro \n";
     auto & warpx = WarpX::GetInstance();
 
-    IntVect Ex_stag = warpx.getEfield_fp(0,0).ixType().toIntVect();
-    IntVect Ey_stag = warpx.getEfield_fp(0,1).ixType().toIntVect();
-    IntVect Ez_stag = warpx.getEfield_fp(0,2).ixType().toIntVect();
-    IntVect Bx_stag = warpx.getBfield_fp(0,0).ixType().toIntVect();
-    IntVect By_stag = warpx.getBfield_fp(0,1).ixType().toIntVect();
-    IntVect Bz_stag = warpx.getBfield_fp(0,2).ixType().toIntVect();
+    // Get BoxArray and DistributionMap of warpx instance.
+    int lev = 0;
+    amrex::BoxArray ba = warpx.boxArray(lev);
+    amrex::DistributionMapping dmap = warpx.DistributionMap(lev);
+    const amrex::IntVect ng_EBSolver = warpx.getngE();
```

Comment:
I wonder if we should define a new "getter" in the `WarpX` class, called `get_ng_FieldSolver` or something like this, which returns the `IntVect` `ng_FieldSolver` of the `guardCellManager` class, and use it here instead of using `getngE` which returns the `IntVect` `ng_alloc_EB`.

I think the two `IntVect`s are probably equal most of the times, but I wonder if there are occasions when they differ and from the name `ng_EBSolver` chosen here it seems like we may want to use `ng_FieldSolver` instead of `ng_alloc_EB`. What do you think?

I'm aware that maybe the name `ng_EBSolver` was chosen not to say that we need to use only the guard cells required for the field solver, but simply that we want to use the guard cells allocated for E and B (to which, collectively and rightfully, we are referring with the label "field solver" because that's the part of the PIC code where they belong the most), not sure about this.

If we don't know yet, it's probably safer to use `getngE` which returns always all guard cells allocated for E and B, as you did, but maybe it could be worth changing the name from `ng_EBSolver` to something like `ng_alloc_EB` (if it doesn't make the compiler complain), so that it's clearer which number of guard cells we are using here. Let me know if this makes sense.

Reply:
getngE() returns ng_alloc_EB, which is what we wanted here. 
I can definitely change this local name to ng_alloc_EB. 
Its just local and a pretty small change, if that helps with clarity.

Reply:
let me know if this is not what you meant?

Reply:
Yes, thanks for the change!